crates/router/src/core/payments/operations/payment_create.rs

```diff
@@ -1426,6 +1427,13 @@ impl PaymentCreate {
             .attach_printable("Unable to encrypt the payment intent data")?;
 
         let skip_external_tax_calculation = request.skip_external_tax_calculation;
+        let tax_amount = request.order_tax_amount.unwrap_or(MinorUnit::new(0));
+        let tax_details = Some(diesel_models::TaxDetails {
```

Comment:
set this only if the value exists in the request. what is happening here is that we are defaulting to a zero value and then always setting this value to default tax, even if the merchant has not passed this.